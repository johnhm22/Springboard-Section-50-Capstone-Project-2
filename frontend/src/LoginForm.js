import React from 'react';
import { useHistory } from "react-router-dom";
import './LoginForm.css';
import { useFormik } from 'formik';


const validate = values => {
    const errors = {};

    if(!values.username) {
        errors.username = "username required";
    } 
    if(!values.password) {
        errors.password = "password required";
    } 
    return errors;
    }

const LoginForm = ({login}) => {

    const history = useHistory();


    const formik = useFormik({
        initialValues: {
            username: '',
            password: '',
        },
        validate,
        onSubmit: values => {
            login(values.username, values.password);
            history.push('/');
        },
    });


return (
    <div className="container loginform col-lg-6">
        <div className="row justify-content-md-center">
        <div className="col-6">
        <form method="get" onSubmit={formik.handleSubmit} className="mt-3">
        <h3>Log In</h3>
            <div className="form-group">
            <label htmlFor="username"></label>
            <input 
            className="form-control" 
            type="text" name="username" 
            id="username" 
            value={formik.username}
            onBlur={formik.handleBlur}
            onChange={formik.handleChange} 
            placeholder="Enter username..." />
            {formik.errors.username ? <div style={{color:'red'}}>{formik.errors.username}</div> : null}
            </div>
            <div className="form-group">
            <label htmlFor="password"></label>
            <input
            className="form-control" 
            type="text" name="password" 
            id="password" 
            value={formik.password} 
            onBlur={formik.handleBlur}
            onChange={formik.handleChange} 
            placeholder="Enter password..." />
            {formik.errors.password ? <div style={{color:'red'}}>{formik.errors.password}</div> : null}
            </div>
            <button type="submit" className="btn btn-primary btn-sm float-right">Log in</button>
        </form>
        </div>
        </div>
    </div>
    )
  
};

export default LoginForm;



