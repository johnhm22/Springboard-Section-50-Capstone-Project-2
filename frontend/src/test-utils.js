import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router
const customRender = (ui, options) => {
  return render(ui, { wrapper: BrowserRouter, ...options });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };