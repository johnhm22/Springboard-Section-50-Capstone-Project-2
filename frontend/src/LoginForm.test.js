import React from 'react';
// import { render } from '@testing-library/react';
import {render} from './test-utils';

import LoginForm from './LoginForm';

//passes
it('renders without crashing', () => {
  render(<LoginForm />);
});

//passes
it('matches snapshot', () => {
    const {asFragment} = render(<LoginForm />);
    expect(asFragment()).toMatchSnapshot();
});
