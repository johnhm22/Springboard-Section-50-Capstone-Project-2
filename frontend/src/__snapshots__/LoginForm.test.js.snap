// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container col-lg-6"
  >
    <div
      class="row justify-content-md-center"
    >
      <div
        class="col-6"
      >
        <form
          class="mt-3"
          method="get"
        >
          <h3>
            Log In
          </h3>
          <div
            class="form-group"
          >
            <label
              for="username"
            />
            <input
              class="form-control"
              id="username"
              name="username"
              placeholder="Enter username..."
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="password"
            />
            <input
              class="form-control"
              id="password"
              name="password"
              placeholder="Enter password..."
              type="text"
              value=""
            />
          </div>
          <button
            class="btn btn-primary btn-sm float-right"
          >
            Log in
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
